pkg/internal/checkup/pod/pod_test.go

```diff
@@ -0,0 +1,193 @@
+package pod_test
+
+import (
+	"testing"
+
+	networkv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kiagnose/kubevirt-dpdk-checkup/pkg/internal/checkup/pod"
+)
+
+const testPodName = "my-pod"
+const testContainerName = "my-container"
+
+func TestCreateNetworkRequestAnnotationShouldSucceed(t *testing.T) {
+	type successTestCase struct {
+		description              string
+		inputNetworkRequestsMap  map[string]*networkv1.NetworkAttachmentDefinition
+		expectedNetworksRequests []string
+	}
+
+	testCases := []successTestCase{
+		{
+			description:              "when the input map is empty",
+			inputNetworkRequestsMap:  map[string]*networkv1.NetworkAttachmentDefinition{},
+			expectedNetworksRequests: []string{""},
+		},
+		{
+			description: "when the input map is not empty",
+			inputNetworkRequestsMap: map[string]*networkv1.NetworkAttachmentDefinition{
+				"AA:AA:AA:AA:AA:01": newNetAttachDef("namespace1", "name1"),
+				"AA:AA:AA:AA:AA:02": newNetAttachDef("namespace2", "name2"),
+			},
+			expectedNetworksRequests: []string{
+				"{\"name\":\"name1\",\"namespace\":\"namespace1\",\"mac\":\"AA:AA:AA:AA:AA:01\",\"cni-args\":null}",
+				"{\"name\":\"name2\",\"namespace\":\"namespace2\",\"mac\":\"AA:AA:AA:AA:AA:02\",\"cni-args\":null}",
+			},
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.description, func(t *testing.T) {
+			actualNetworksRequest, err := pod.CreateNetworksRequest(testCase.inputNetworkRequestsMap)
+			assert.NoError(t, err)
+			for _, expectedRequest := range testCase.expectedNetworksRequests {
+				assert.Contains(t, actualNetworksRequest, expectedRequest)
+			}
+		})
+	}
+}
+
+func newNetAttachDef(namespace, name string) *networkv1.NetworkAttachmentDefinition {
+	return &networkv1.NetworkAttachmentDefinition{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+		},
+	}
+}
```

Comment:
Could you please move this helper to the bottom of the file?

Reply:
DONE